File: Minecraft2D/Entity/EntityPlayer.cs
using System;
using System.Drawing;
using System.Runtime.CompilerServices;
using System.Threading.Tasks;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace Minecraft2D
{
    public class EntityPlayer
    {
        public string Name { get; set; }
        public string UUID { get; set; }
        public string ItemInHand { get; set; } = "";
        public Image ItemImage { get; set; }
        public Image ItemImageFlipped { get; set; }
        public Image Sprite { get; set; } = My.MyProject.Forms.Form1.playerSkin;
        public Point Location { get; set; }
        public TransparentPicBox HandItem { get; set; }
        public int LastWalk { get; set; } = 0;

        private TransparentPicBox _Render;

        public TransparentPicBox Render
        {
            [MethodImpl(MethodImplOptions.Synchronized)]
            get
            {
                return _Render;
            }

            [MethodImpl(MethodImplOptions.Synchronized)]
            internal set
            {
                if (_Render != null)
                {
                    _Render.LocationChanged -= (_, __) => Test();
                }

                _Render = value;
                if (_Render != null)
                {
                    _Render.LocationChanged += (_, __) => Test();
                }
            }
        }

        public EntityPlayer(string arg0, string arg1, Point arg2, TransparentPicBox arg3)
        {
            Name = arg0;
            UUID = arg1;
            Location = arg2;
            Render = arg3;
            HandItem = new TransparentPicBox()
            {
                Name = "Testrender",
                BackColor = Color.Transparent,
                SizeMode = PictureBoxSizeMode.StretchImage
            };
            My.MyProject.Forms.Form1.Controls.Add(HandItem);
        }

        private void Test()
        {
            if (LastWalk == 1)
            {
                Render.Image = Sprite;
            }
            else
            {
                Render.Image = Form1.GetInstance().GetFlipped(Sprite);
            }

            try
            {
                if (Information.IsNothing(ItemImageFlipped))
                {
                    HandItem.Hide();
                    return;
                }

                if (Information.IsNothing(ItemImage))
                {
                    HandItem.Hide();
                    return;
                }

                HandItem.Show();
                var lc = Render.Location;
                if (ItemImage.Equals(null))
                    return;
                if (LastWalk == 1)
                {
                    lc.X += Render.Width - 5;
                    HandItem.Image = ItemImage;
                }
                else
                {
                    lc.X -= HandItem.Width - 5;
                    HandItem.Image = ItemImageFlipped;
                }

                lc.Y = (int)(lc.Y + (45d - HandItem.Height / 2d));
                HandItem.Size = new Size(24, 24);
                HandItem.SizeMode = PictureBoxSizeMode.StretchImage;
                HandItem.BringToFront();
                HandItem.Location = lc;
            }
            catch (Exception)
            {
                HandItem.Hide();
            }
        }

        public async Task SetItemInHand(Image i, Image iflipped, string str)
        {
            if (!Information.IsNothing(i))
                HandItem.Image = i;
            ItemImage = i;
            ItemImageFlipped = iflipped;
            ItemInHand = str;
            Render.Update();
            HandItem.Update();
        }

        public void Remove()
        {
            if (My.MyProject.Forms.Form1.InvokeRequired)
            {
                My.MyProject.Forms.Form1.Invoke(new MethodInvoker(Remove));
            }
            else
            {
                My.MyProject.Forms.Form1.Controls.Remove(HandItem);
            }
        }
    }
}

File: Minecraft2D/Entity/EntityPlayer.vb
Public Class EntityPlayer
    Property Name As String
    Property UUID As String
    Property ItemInHand As String = ""
    Property ItemInImage As Image
    Property ItemInImageFlipped As Image
    Property Location As Point
    Property R1 As TransparentPicBox
    Property LastWalk As Integer = 0
    WithEvents Render As PictureBox
    Public Sub New(arg0 As String, arg1 As String, arg2 As Point, arg3 As PictureBox)
        Name = arg0
        UUID = arg1
        Location = arg2
        Render = arg3

        R1 = New TransparentPicBox With {
        .Name = "Testrender",
        .BackColor = Color.Transparent,
        .SizeMode = PictureBoxSizeMode.StretchImage
        }

        Form1.Controls.Add(R1)
    End Sub
    Private Sub Test() Handles Render.LocationChanged
        If LastWalk = 1 Then
            Render.Image = Form1.playerSkin
        Else
            Render.Image = Form1.playerSkinFlip
        End If
        Try
            If IsNothing(ItemInImageFlipped) Then
                R1.Hide()
                Exit Sub
            End If
            If IsNothing(ItemInImage) Then
                R1.Hide()
                Exit Sub
            End If
            R1.Show()
            Dim lc = Render.Location
            If ItemInImage.Equals(Nothing) Then Exit Sub
            If LastWalk = 1 Then
                lc.X += Render.Width - 10
                R1.Image = ItemInImage
            Else
                lc.X -= R1.Width - 10
                R1.Image = ItemInImageFlipped
            End If
            lc.Y += 55 - (R1.Height / 2)
            R1.Size = New Size(32, 32)
            R1.SizeMode = PictureBoxSizeMode.StretchImage
            R1.BringToFront()

            R1.Location = lc
        Catch ex As Exception
            R1.Hide()
        End Try
    End Sub
    Sub SetItemInHand(i As Image, iflipped As Image, str As String)
        If Not IsNothing(i) Then R1.Image = i
        ItemInImage = i
        ItemInImageFlipped = iflipped
        ItemInHand = str
        Render.Update()
        R1.Update()
    End Sub

    Sub Remove()
        If Form1.InvokeRequired Then
            Form1.Invoke(New MethodInvoker(AddressOf Remove))
        Else
            Form1.Controls.Remove(R1)
        End If
    End Sub

End Class
